mkosi/machine.py

```diff
@@ -122,9 +127,44 @@ def run(self, commands: Sequence[str], timeout: int = 900, check: bool = True) -
 
         return process
 
-    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+    def _kill(self, *args: Any, **kwargs: Any) -> None:
         if self._serial:
             self._serial.kill(signal.SIGTERM)
             self.exit_code = self._serial.wait()
             self._serial = None
         self.stack.__exit__(*args, **kwargs)
+
+    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+        self._kill(*args, **kwargs)
+
+    def reboot(self, *args: Any, **kwargs: Any) -> None:
+        self._kill(*args, **kwargs)
```

Comment:
This signature for reboot() is quite confusing. It's hard to see from the signature what arguments we are supposed to pass to it.

I'd go for a different solution where if reboot() is called manually, let's not pass any exception information to `_kill`. So remove the arguments from reboot() and have it call `_kill` like so: `self._kill(None, None, None)`

Reply:
This may have broken the exit of qemu, will try again later to confirm it